pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -286,18 +224,22 @@ func (am *alertmanager) aggregateInhibitMatchers(rules []config.InhibitRule, amu
 // applyConfig applies a new configuration by re-initializing all components using the configuration provided.
 // It returns a boolean indicating whether the user config was changed and an error.
 // It is not safe to call concurrently.
-func (am *alertmanager) applyConfig(cfg *apimodels.PostableUserConfig, rawConfig []byte) (bool, error) {
-	// First, let's make sure this config is not already loaded
-	var amConfigChanged bool
-	if rawConfig == nil {
-		enc, err := json.Marshal(cfg.AlertmanagerConfig)
-		if err != nil {
-			// In theory, this should never happen.
-			return false, err
-		}
-		rawConfig = enc
+func (am *alertmanager) applyConfig(config *ngmodels.AlertConfiguration) (bool, error) {
+	// Convert to PostableApiAlertingConfig, which is the format used by the GrafanaAlertmanager.
+	cfg, err := Load([]byte(config.AlertmanagerConfiguration))
+	if err != nil {
+		return false, err
 	}
 
+	rawConfig, err := json.Marshal(cfg.AlertmanagerConfig)
+	if err != nil {
+		return false, err
+	}
+
+	// Ensure this config is not already loaded
+	var amConfigChanged bool
+	// Note, the hash saved to the database in config.ConfigurationHash (PostableUserConfig) is different from the one
```

Comment:
In some cases, marshalling a configuration can result in different JSON than the original JSON that was read. For example, matchers that were `foo=bar` will be marshalled as `foo="bar"`, and as such the hashes won't match. This won't be an issue here, right?

Reply:
I don't believe it will be an issue. For the purposes of the internal AM the hashes just have to be consistent. So at worst, this could cause a single extra apply if an existing database configuration has a discrepancy from a difference in marshalling.

Overall, I expect this change to either reduce the amount of unnecessary updates or remain the same, given the amount of conditional complexity that's been removed in how we create the rawConfig/hash .